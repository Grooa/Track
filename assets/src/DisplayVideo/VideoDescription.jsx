import PropTypes from 'prop-types';
import ReactMarkdown from 'react-markdown';

const VideoDescription = ({ description = '', className = '' }) => <section className={className}>
  <h2>Description</h2>

  {!!description && <ReactMarkdown className="data" source={description} escapeHtml={false} />}
</section>;

VideoDescription.propTypes = {
  description: PropTypes.string,
  className: PropTypes.string,
};

export default VideoDescription;
