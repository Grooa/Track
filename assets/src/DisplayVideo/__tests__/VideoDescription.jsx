import { render, shallow } from 'enzyme';

import VideoDescription from '../VideoDescription';

describe('<VideoDescription />', () => {
  let description;
  let className;

  beforeEach(() => {
    description = '# Hello world \n This is a paragraph';
    className = 'test-test some-css-class';
  });

  it('should render empty description properly', () => {
    const wrapper = render(<VideoDescription />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render description properly', () => {
    const wrapper = render(<VideoDescription description={description} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should not render ReactMarkdown when empty', () => {
    const wrapper = render(<VideoDescription />);

    expect(wrapper.find('ReactMarkdown').length).toBe(0);
  });

  it('should render ReactMarkdown when not empty', () => {
    // Must use shallow() here, as render() will tell ReactMarkdown to run.
    const wrapper = shallow(<VideoDescription description={description} />);

    expect(wrapper.find('ReactMarkdown').length).toBe(1);
  });

  it('should place className on <section>', () => {
    const wrapper = shallow(<VideoDescription className={className} />);

    expect(wrapper.find('section').hasClass(className)).toBeTruthy();
  });
});
