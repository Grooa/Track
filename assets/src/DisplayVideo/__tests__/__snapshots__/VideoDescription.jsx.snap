// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoDescription /> should render description properly 1`] = `
<section
  class=""
>
  <h2>
    Description
  </h2>
  <div
    class="data"
  >
    <h1>
      Hello world
    </h1>
    <p>
       This is a paragraph
    </p>
  </div>
</section>
`;

exports[`<VideoDescription /> should render empty description properly 1`] = `
<section
  class=""
>
  <h2>
    Description
  </h2>
</section>
`;
